server.js

```diff
@@ -415,14 +418,39 @@ function addGHHook(app, path) {
 
             var owner = event.repository.owner.login
             ,   repo = event.repository.full_name
+            ,   repoShortname = event.repository.name
             ,   repoId = event.repository.id
             ;
-            store.getSecret(repo, function (err, data) {
-                // if there's an error, we can't set an error on the status because we have no secret, so bail
-                if (err || !data) return error(res, "Secret for " + repo + " not found: " + (err || "simply not there."));
+            store.getSecret(repo, async function (err, data) {
+                if (err || !data) {
+                    try {
+                        const { token } = await doAsync(store).getToken(owner);
+                        const gh = new GH({ accessToken: token });
+                        const statusData = {
+                            owner,
+                            shortName: repoShortname,
+                            sha: event.pull_request.head.sha,
+                            payload: {
+                                state: "failure",
+                                target_url: `${config.url}pr/id/${owner}/${repoShortname}/${event.number}`,
+                                description: `The repository manager doesn't know the following repository: ${repo}`,
+                                context: "ipr"
+                            }
+                        };
+                        gh.status(statusData, err => {
+                            if (err) {
+                                console.log(err);
+                                log.error(err);
+                            }
+                        });
+                    } catch (e) {
+                        log.error(`Token not found for ${owner}`);
+                    }
+                    return error(res, "Secret for " + repo + " not found: " + (err || "simply not there."));
+                }
 
                 // we have the secret, crypto check becomes possible
-                if (!GH.checkPayloadSignature("sha1", data.secret, buffer, req.headers["x-hub-signature"])) return error(res, "GitHub signature does not match known secret for " + repo + ".");
+                if (!GH.checkPayloadSignature("sha256", data.secret, buffer, req.headers["x-hub-signature"])) return error(res, "GitHub signature does not match known secret for " + repo + ".");
```

Comment:
can you say more about the change from sha1 to sha256? Is it backwards compatible with our existing data?

Reply:
more specifically, if I understand https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks#validating-payloads-from-github correctly, to use sha-256 we would need to take the signature from X-Hub-Signature-256

Reply:
> more specifically, if I understand https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks#validating-payloads-from-github correctly, to use sha-256 we would need to take the signature from X-Hub-Signature-256

That's correct, I forgot to update the header to be checked.
I started looking at the signature verification to see if the second issue you reported at https://github.com/w3c/ash-nazg/issues/210 wasn't related to that function. It doesn't seem to be the case but we could still follow the GH recommendation of using the SHA-256 signature. I understand this shouldn't break the existing data and the function will only check future payloads.